sched/pthread/pthread_exit.c

```diff
@@ -87,12 +87,6 @@ void pthread_exit(FAR void *exit_value)
   tcb->cpcount = 0;
 #endif
 
-#ifdef CONFIG_PTHREAD_CLEANUP
-  /* Perform any stack pthread clean-up callbacks */
-
-  pthread_cleanup_popall(tcb);
```

Comment:
pthread_cleanup should not be called from kernel side, we should resolve it in later patch